app/scripts/ChromeSerialBridge.js

```diff
@@ -96,13 +97,15 @@ export default class ChromeSerialBridge {
         }
 
         var toSend = queue.shift();
+        sendPending = true;
         serialPort.write(toSend, function (err, results) {
+          sendPending = false;
           if (results.error) {
             queue.unshift(toSend);
           }
 
           if (queue.length !== 0) {
-            setTimeout(trySend, 0);
+            trySend();
```

Comment:
Why not setTimeout 0 anymore? Because we're now waiting for confirmation of each send so we're not worried about getting caught in a recursive trySend without completing the sends?

Reply:
Yeah, I think I was wrong here:

> My reason for adding it was to make sure we don't get caught in recursive trySend() calls while never pausing to allow the send to complete (in case it can get blocked by the JS thread).

We're already inside the `serialPort.write` callback here, so the browser is yielded to between each write call.